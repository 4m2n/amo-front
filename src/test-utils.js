import {
  createStore,
} from "redux"
import {
  Provider,
} from "react-redux"
import {
  TestScheduler,
} from "rxjs/testing"
import rootReducer from "./state"
import React from "react"

// done :: () -> _
export const done = () => undefined

// createContainer :: React.Component -> React.Component
export const createContainer = component =>
  <Provider store={createStore(rootReducer)}>
    {component}
  </Provider>

// createTestScheduler :: () -> TestScheduler
export const createTestScheduler = () => new TestScheduler((actual, expected) => {
  expect(actual).toEqual(expected)
})
