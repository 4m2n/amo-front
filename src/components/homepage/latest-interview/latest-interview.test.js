import { render } from "@testing-library/react"
import renderer from "react-test-renderer"
import { pipe, prop, tap } from "ramda"
import { createContainer, done } from "./../../../test-utils"
import LatestInterview from "./latest-interview"
import React from "react"

describe("components :: homepage :: latest-interview", () => {
  it("displays a loader while loading the latest interview", () => pipe(
    props => <LatestInterview {...props} />,
    createContainer,
    render,
    prop("container"),
    container => container.querySelector("p"),
    loader => expect(loader.textContent).toBe("Chargement en cours..."),
  )({
    isReady: false,
    interview: null,
  }))

  it("displays a message in case, after loading, there is no interview to display", () => pipe(
    props => <LatestInterview {...props} />,
    createContainer,
    render,
    prop("container"),
    container => container.querySelector("p"),
    noNews => expect(noNews.textContent).toBe("Pas de nouveautÃ©s en ce moment !"),
  )({
    isReady: true,
    interview: null,
  }))

  it("displays the interview content after loading", () => pipe(
    props => <LatestInterview {...props} />,
    createContainer,
    renderer.create,
    component => component.toJSON(),
    tap(tree => expect(tree).toMatchSnapshot()),
    done,
  )({
    isReady: true,
    interview: {
      artwork_url: "my-wonderfull-artwork",
      created_at: "2011-10-05T14:48:00.000Z",
      description: "A test interview",
      permalink_url: "this-links-to-soundcloud",
    },
  }))
})
