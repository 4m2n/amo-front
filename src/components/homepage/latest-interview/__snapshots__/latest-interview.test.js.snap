// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components :: homepage :: latest-interview displays the interview content after loading 1`] = `
<div
  className="latest-interview"
>
  <h1
    className="title"
  >
    Dernières actualités
  </h1>
  <div
    className="soundcloud-widget"
  >
    <iframe
      allow="autoplay"
      className="is-hidden"
      id="sc-interview"
      src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/playlists/1290749911"
      title="sc-interview"
    />
  </div>
  <figure>
    <img
      alt=""
      src="my-wonderfull-artwork"
    />
    <figcaption>
      <div
        className="soundcloud-controls "
      >
        <div
          className="loader"
        />
      </div>
    </figcaption>
  </figure>
  <time>
    5 octobre 2011
  </time>
  <p
    className="description"
  >
    A test interview
  </p>
</div>
`;
